components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/common/InboundRequestProcessorImpl.java

```diff
@@ -68,47 +70,78 @@ public InboundRequestProcessorImpl() {
     protected void start(InboundTask task, String endpointPostfix) {
         log.info("Starting the inbound endpoint [" + name + "]" + (startInPausedMode ? " in suspended mode" : "")
                 + ", with coordination " + coordination + ". Interval : " + interval + ". Type : " + endpointPostfix);
+        //This only supports Generic tasks for cron expression based scheduling
+        if (task instanceof GenericTask)  {
+            cronExpression = ((GenericTask) task).getPollingConsumer().getCronExpression();
+        }
         if (coordination) {
-            try {
-                TaskDescription taskDescription = new TaskDescription();
-                taskDescription.setName(name + "-" + endpointPostfix);
-                taskDescription.setTaskGroup(endpointPostfix);
-                if (interval < InboundTask.TASK_THRESHOLD_INTERVAL) {
-                    taskDescription.setInterval(InboundTask.TASK_THRESHOLD_INTERVAL);
-                } else {
-                    taskDescription.setInterval(interval);
-                }
-                taskDescription.setIntervalInMs(true);
-                taskDescription.addResource(TaskDescription.INSTANCE, task);
-                taskDescription.addResource(TaskDescription.CLASSNAME, task.getClass().getName());
-                taskDescription.setTaskImplClassName(task.getClass().getName());
-                taskDescription.addProperty(TaskUtils.TASK_OWNER_PROPERTY, TaskUtils.TASK_BELONGS_TO_INBOUND_ENDPOINT);
-                taskDescription.addProperty(TaskUtils.TASK_OWNER_NAME, name);
-                taskDescription.addProperty(TaskUtils.START_IN_PAUSED_MODE, String.valueOf(startInPausedMode));
-                StartUpController startUpController = new StartUpController();
-                startUpController.setTaskDescription(taskDescription);
-                startUpController.init(synapseEnvironment);
-                startUpControllersList.add(startUpController);
-                // Register a listener to be notified when the local JMS/Generic task is deleted/paused
-                if (task instanceof JMSTask || task instanceof GenericTask) {
-                    TaskManager taskManagerImpl = synapseEnvironment.getTaskManager().getTaskManagerImpl();
-                    if (taskManagerImpl instanceof NTaskTaskManager) {
-                        NTaskTaskManager ntaskManager = (NTaskTaskManager) taskManagerImpl;
-                        if (task instanceof JMSTask) {
-                            ntaskManager.registerListener((JMSTask) task, taskDescription.getName());
-                        } else if (task instanceof GenericTask) {
-                            ntaskManager.registerListener((GenericTask) task, taskDescription.getName());
-                        }
+            handleTask(task, endpointPostfix);
+        } else if (cronExpression != null && !cronExpression.isEmpty()) {
```

Comment:
you can simplify this if, else-if block with a single if block as below.

if (coordination || (cronExpression != null && !cronExpression.isEmpty())) {
         handleTask(task, endpointPostfix);
}

WDYT?

Reply:
Yeah , I will change